Sources/Socket.swift

```diff
@@ -3128,191 +3123,131 @@ public class Socket: SocketReader, SocketWriter {
 	///
 	/// - Returns: number of bytes read.
 	///
-	private func readDataIntoStorage() throws -> Int {
+	private func readDataIntoStorage() throws -> (bytesRead: Int, address: Address?) {
 		
 		// Clear the buffer...
 		self.readBuffer.deinitialize()
 		self.readBuffer.initialize(to: 0x0)
 		memset(self.readBuffer, 0x0, self.readBufferSize)
-		
+		var address: Address? = nil
+
+		guard let sig = self.signature else {
+			throw Error(code: Socket.SOCKET_ERR_INTERNAL, reason: "Socket signature not found.")
+		}
+
+		let addr = sockaddr_storage()
+		var length = socklen_t(MemoryLayout<sockaddr_storage>.size)
+		var addrPtr = addr.asAddr()
+
 		// Read all the available data...
 		var count: Int = 0
 		repeat {
-			
+
 			if self.delegate != nil {
```

Comment:
You missed a *key difference* between `readDatagramIntoStorage` and `readDataIntoStorage`.  SSL is not supported (**YET**) for UDP.  This requires use **DTLS* which the current implementation of *BlueSSLService* does not implement...